### ISSUE TITLE
Safe handling of `Self` typed attributes w.r.t. overrides.

### ISSUE THREAD

**Issue by ilevkivskyi** [python/mypy#14074]
This is a follow up for https://github.com/python/mypy/pull/14041

This example doesn't give an error, while it probably should:
```python
from typing import Self

class C:
    x: Self

class D(C):
    x: D  # Note this is already unsafe, but it essentially same as #2353 (for methods)

class E(D):
    pass  # This creates conflicting types for `x` from `C` and from `D`.
```
